contrib/pyln-client/pyln/client/plugin.py

```diff
@@ -37,6 +37,11 @@ class RequestState(Enum):
     FINISHED = 'finished'
     FAILED = 'failed'
 
+class CLNRestData(TypedDict):
+    path: str
+    method: str
```

Comment:
Adding path and method seems useless as these cannot be changed for clnrest. Path will always be the name of the rpc command (`@rpcns.route("/<rpc_method>")`) and method is always POST (`def post(self, rpc_method)`).  We can add GET, DELETE etc. methods types in clnrest but it seems unnecessary because POST can be used for all types.

Reply:
I was thinking clnrest could be modified to register path and method dynamically. Already been talking to @daywalker90 about adding that feature to his rust version of clnrest (#7509)
This is why I also opened #7508, so that clnrest could dynamically add routes as new plugins start, and remove those routes when plugins stop.
True that POST can be used for all types, but a plugin may want to implement a specific protocol that defines what the method should be for some endpoint. Note "Motivations" above...